docs/development.md

```diff
@@ -35,3 +35,39 @@ Since this is very unlikely and most times an abs path is not strictly necessary
 "Should" functions can be used anywhere,
 but care should be taken not to overuse them when a fallback value is not sufficient.
 "Must" variants should be favored in testing code, since they will more obviously fail tests.
+
+## Cross-Platform Scripting
+
+We are writing bash scripts with the intention of running them in a POSIX environment.
+However, differences between operating systems, particularly between Linux and MacOS,
+can still affect a script's behavior. 
+This section highlights key considerations to ensure your scripts run consistently across platforms.
+
+### Regular Expressions
+Differences between how Linux and MacOS handle regular expressions can cause inconsistent outcomes when writing scripts.
+Be aware of the following:
+
+- **BRE (Basic Regular Expressions)**
+  - **Usage**: Default in many tools.
+  - **Consideration**: Limited feature set.
+  - **[More Info](https://en.wikibooks.org/wiki/Regular_Expressions/Basic_Regular_Expressions)**
```

Comment:
There is something missing here.
If this was fully true, then we would just always use BREs.
The functionality may be "limited" (I would never say that about the core set of Regex features),
but they are almost always all we need (easily 98% of cases would be solved with BREs).

So, why don't we just only use BREs?